src/posix/platform/infra_if.cpp

```diff
@@ -250,6 +250,9 @@ void platformInfraIfInit(otInstance *aInstance, const char *aIfName)
         DieNow(OT_EXIT_ERROR_ERRNO);
     }
 
+    rval = setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE, sInfraIfName, strlen(sInfraIfName));
+    verifyOrDie(rval == 0, OT_EXIT_ERROR_ERRNO);
```

Comment:
follow the same code pattern as above?

```suggestion
    rval = setsockopt(sock, SOL_SOCKET, SO_BINDTODEVICE, sInfraIfName, strlen(sInfraIfName));
    if (rval < 0) 
    {
        otLogCritPlat("Can't set SO_BINDTODEVICE: %s", strerror(errno));
        DieNow(OT_EXIT_ERROR_ERRNO);
    }
```

Reply:
We are simplifying the infra IF logging: https://github.com/openthread/openthread/pull/6090